app/models/quarterly.rb

```diff
@@ -0,0 +1,15 @@
+class Quarterly
+  # transform valid month into quarter
+  def initialize(month)
+    @month = month
+  end
+
+  def to_quarter
+    (month_index / 3.0).ceil
+  end
+
+  private
+    def month_index
+      I18n.t(:abbr_month_names, scope: :date).index(@month) || raise("#{@month} is not a valid month")
```

Comment:
@radinreth be careful of this raise, I didn't see anywhere catch this exception, it means that if the request encounter this exception -> it will break a whole process but not just ignore the relevant chart

Reply:
I raise the exception here, in case there are somewhere else using this Quarterly class.
but in this case i catch the exception because it is from rails, strftime, so it's must be a valid date

Reply:
anyway, what is your recommendation bong?

Reply:
@radinreth we should catch this exception to let dashboard load the rest visualization while the one encountered exception could be a message "Problem with data loading" and ensure this will capture by Sentry where we will get the alert notification in a timely manner.

Reply:
@kakada , pls check again bong